minethd.cpp

```diff
@@ -33,7 +33,11 @@
 
 void thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id)
 {
+#ifdef __MINGW64__
+	SetThreadAffinityMask(&h, 1ULL << cpu_id);
```

Comment:
6.2.0\x86_64-w64-mingw32\include\winbase.h
<pre>
1160  WINBASEAPI DWORD_PTR WINAPI SetThreadAffinityMask (HANDLE hThread, DWORD_PTR dwThreadAffinityMask);
</pre>

6.2.0\x86_64-w64-mingw32\include\winnt.h
<pre>
250   typedef void *PVOID;
251   typedef void *PVOID64;
</pre>

6.2.0\x86_64-w64-mingw32\include\winnt.h (line 397 - 403)
<pre>
#ifdef STRICT
  typedef void *HANDLE;
#define DECLARE_HANDLE(name) struct name##__ { int unused; }; typedef struct name##__ *name
#else
typedef PVOID HANDLE;
#define DECLARE_HANDLE(name) typedef HANDLE name
#endif
</pre>

Therefore SetThreadAffinityMask is just a: 
<pre>WINBASEAPI DWORD_PTR WINAPI SetThreadAffinityMask (void* hThread, DWORD_PTR dwThreadAffinityMask);
</pre>

Hence, nothing goes into **hThread**, except  **ptr variable** or an **address of a (lvalue) variable.**

Thanks